src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs

```diff
@@ -29,6 +29,8 @@ internal sealed class SocketConnection : TransportConnection
 
         private volatile bool _aborted;
 
+        private SequencePosition SendEnd;
```

Comment:
This is pretty gross, can this be left as a local (that will get hoisted).

Reply:
Yeah :-/

But it means the main loop doesn't need error handling; so there is only the try catch from the statemachine itself. 

Putting it back in the loop means the `finally` has to go back to ensure `Output.AdvanceTo(End);` is always called; even when its crashing out.

Reply:
Is a consequence of not doing https://github.com/dotnet/corefx/issues/27465?

Reply:
Oh... maybe its not needed? https://github.com/dotnet/corefx/pull/27596

Reply:
Removed